pallets/msa/src/weights.rs

```diff
@@ -88,87 +88,88 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: Msa DelegatorAndProviderToDelegation (r:1 w:1)
 	// Storage: Schemas CurrentSchemaIdentifierMaximum (r:1 w:0)
 	fn create_sponsored_account_with_delegation() -> Weight {
-		Weight::from_ref_time(100_890_000 as u64)
+		Weight::from_ref_time(103_320_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(8 as u64))
 			.saturating_add(T::DbWeight::get().writes(5 as u64))
 	}
 	// Storage: Msa PublicKeyToMsaId (r:1 w:0)
 	// Storage: Msa DelegatorAndProviderToDelegation (r:1 w:1)
 	fn revoke_delegation_by_provider(s: u32, ) -> Weight {
-		Weight::from_ref_time(49_497_000 as u64)
+		Weight::from_ref_time(48_703_000 as u64)
 			// Standard Error: 2_000
-			.saturating_add(Weight::from_ref_time(78_000 as u64).saturating_mul(s as u64))
+			.saturating_add(Weight::from_ref_time(79_000 as u64).saturating_mul(s as u64))
 			.saturating_add(T::DbWeight::get().reads(2 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: Msa PayloadSignatureRegistry (r:2 w:2)
 	// Storage: Msa PublicKeyToMsaId (r:2 w:1)
 	// Storage: Msa PublicKeyCountForMsaId (r:1 w:1)
 	fn add_public_key_to_msa() -> Weight {
-		Weight::from_ref_time(147_481_000 as u64)
+		Weight::from_ref_time(146_185_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(5 as u64))
 			.saturating_add(T::DbWeight::get().writes(4 as u64))
 	}
 	// Storage: Msa PublicKeyToMsaId (r:2 w:1)
 	// Storage: Msa PublicKeyCountForMsaId (r:1 w:1)
 	fn delete_msa_public_key() -> Weight {
-		Weight::from_ref_time(30_493_000 as u64)
+		Weight::from_ref_time(30_196_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(3 as u64))
 			.saturating_add(T::DbWeight::get().writes(2 as u64))
 	}
 	// Storage: Msa PublicKeyToMsaId (r:1 w:1)
 	// Storage: Msa PublicKeyCountForMsaId (r:1 w:1)
+	// Storage: Msa DelegatorAndProviderToDelegation (r:0 w:126)
 	fn retire_msa() -> Weight {
-		Weight::from_ref_time(31_300_000 as u64)
+		Weight::from_ref_time(186_146_000 as u64)
 			.saturating_add(T::DbWeight::get().reads(2 as u64))
-			.saturating_add(T::DbWeight::get().writes(2 as u64))
+			.saturating_add(T::DbWeight::get().writes(128 as u64))
```

Comment:
@enddynayn writes here should be dependent on a param not being `128` on every call

Reply:
Yeah. we probably do not want this.